creme/geolocation/registry.py

```diff
@@ -0,0 +1,42 @@
+import logging
+from dataclasses import dataclass
+
+from django.db.models import Model
+
+from creme.creme_core.utils.media import creme_media_themed_url
+
+logger = logging.getLogger(__name__)
+
+
+# TODO : Add more properties here like icon size or shadow.
+@dataclass
+class GeoMarkerIcon:
+    path: str = ''
+
+    @property
+    def url(self):
+        try:
+            return creme_media_themed_url(self.path) if self.path else ''
+        except KeyError:
+            logger.warning(f'Missing marker image: {self.path}')
+            return ''
+
+
+class GeoMarkerIconRegistry:
+    __slots__ = ('_icons',)
+
+    def __init__(self) -> None:
+        self._icons: dict[type[Model], GeoMarkerIcon] = {}
+
+    def register(self, model: type[Model], icon: GeoMarkerIcon | str) -> "GeoMarkerIconRegistry":
```

Comment:
`from __future__ import annotations` to avoid using a string for static typing

Reply:
:+1: